users/models.py

```diff
@@ -103,3 +103,31 @@ def soft_delete(self: "User"):
         self.posts.update(curation_status=Post.CurationStatus.DELETED)
 
         self.save()
+
+
+class UserCampaignRegistration(TimeStampedModel):
+    """
+    This model stores registration details for a user as part of an
+    campaign/initiative (e.g.: joint project with an external party).
+    It includes a reference to the user, a JSON field for extra details,
+    and key for tracking registration initiatives/campaigns.
+    """
+
+    user = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="registration_details"
+    )
+
+    details = models.JSONField(default=dict, blank=True, null=False)
+
+    key = models.CharField(
+        max_length=200,
+        blank=True,
+        null=True,
```

Comment:
Let's use `null=False` for required Char values to avoid headache with null vs empty string states